ambari-server/src/main/resources/stack-hooks/before-START/scripts/params.py

```diff
@@ -316,7 +316,7 @@
   namenode_rpc = module_configs.get_property_value(module_name, 'hdfs-site', 'dfs.namenode.rpc-address', default_fs)
 
 # if HDFS is not installed in the cluster, then don't try to access namenode_rpc
-if has_namenode and namenode_rpc and 'core-site' in module_configs:
+if has_namenode and namenode_rpc and module_configs.get_all_properties(module_name, 'core-site') is not None:
```

Comment:
get_all_properties() never returns None. It returns an empty map if the config-type does not exist in the command.json